### ISSUE TITLE
Big wish: Allow Avisynth script processing

### ISSUE THREAD

**Issue by Ryrynz** [zachsaw/MPDN_Extensions#177]
Give ffdshow the boot.
Allow scripts to be run based on conditionals.

**Comment by Shiandow** [comment]
Just to clarify, the "big wish" label doesn't give issues more priority. The order is something like:

wontfix < big wish < enhancement < todo < bug

**Comment by zachsaw** [comment]
Actually do you know what's required to support Avisynth scripts? Is there a Directshow source filter that loads Avisynth scripts?

**Comment by Ryrynz** [comment]
Yeah, I gathered the priority order hence the label, I don't expect this one to be tackled any time soon, but maybe it's not that hard since Potplayer added support for it (with some limitations)

http://forum.videohelp.com/threads/362476-Potplayer-and-AviSynth

I'm not aware of anything other than ffdshow's Raw filter and Potplayer that are capable of working with Avisynth. Certainly support for it would fit nicely with MPDN's bonanza of picture processing options.

**Comment by zachsaw** [comment]
LAV Source Splitter opens avs files doesn't it?

**Comment by Ryrynz** [comment]
Oh yes, it can :D

**Comment by zachsaw** [comment]
Uh so what else is there to support?

**Comment by Ryrynz** [comment]
Profile management.

**Comment by Shiandow** [comment]
So basically a player extension is needed to control avisynth?

**Comment by Ryrynz** [comment]
Yeah, like ffdshow raw list of all scripts and allow auto load conditions.

**Comment by Shiandow** [comment]
Depending on how easy it is to insert avisynth in the filter chain, that would be doable.

**Comment by zachsaw** [comment]
It's not going to be possible at all actually - Avisynth lives on the CPU. Every single frame needs to be transferred back to the system memory and then back again to the GPU. The only thing that could be done is a completely different profiling system for Avisynth.

**Comment by Shiandow** [comment]
I thought we were just talking about preprocessing using avisynth? But inserting it somewhere in the middle of the processing would indeed be very difficult.
